multicluster/controllers/multicluster/core/resourceimport_controller.go

```diff
@@ -160,25 +174,27 @@ func (r *ResourceImportReconciler) handleResImpUpdateForService(ctx context.Cont
 		return ctrl.Result{}, err
 	}
 	svcImpObj := getMCServiceImport(resImp)
+	if !reflect.DeepEqual(*svc, corev1.Service{}) {
+		svcImpObj.Spec.IPs = []string{svc.Spec.ClusterIP}
+	}
 	if apierrors.IsNotFound(err) {
-		creationSucceed = false
 		err := r.localClusterManager.Create(ctx, svcImpObj, &client.CreateOptions{})
 		if err != nil {
 			klog.ErrorS(err, "fail to create ServiceImport", "serviceimport", klog.KObj(svcImpObj))
 			return ctrl.Result{}, err
 		}
-		creationSucceed = true
+		svcImpCreated = true
 	}
 
-	if creationSucceed {
+	if svcCreated && svcImpCreated {
```

Comment:
Should we check both, or just svcImpCreated? At least for r.installedResImports.Add(*resImp) we should not care about svcCreated?

Reply:
fix via check `svcImpNotFound` only